sql/src/main/java/io/crate/exceptions/Exceptions.java

```diff
@@ -76,6 +77,34 @@ public static Throwable unwrap(@Nonnull Throwable t) {
         return result;
     }
 
+    public static Throwable unwrap(@Nonnull Throwable t, Class... classes) {
+        int counter = 0;
+        Throwable result = t;
+        while (((BooleanSupplier) () -> {
```

Comment:
Please move the BooleanSupplier definition outside and assign it to a variable. Otherwise it's re-created for each iteration